diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-03-19 10:39:31.000000000 +0100
+++ b/CMakeLists.txt	2024-03-18 05:20:45.000000000 +0100
@@ -11,8 +11,8 @@
 
 project(telly-skout VERSION ${RELEASE_SERVICE_VERSION})
 
-set(KF_MIN_VERSION "5.75.0")
-set(QT_MIN_VERSION "5.15.0")
+set(KF_MIN_VERSION "6.0.0")
+set(QT_MIN_VERSION "6.6.0")
 
 include(FeatureSummary)
 
diff -urN a/.gitlab-ci.yml b/.gitlab-ci.yml
--- a/.gitlab-ci.yml	2024-03-19 10:39:31.000000000 +0100
+++ b/.gitlab-ci.yml	2024-03-18 05:20:45.000000000 +0100
@@ -6,12 +6,8 @@
     file:
       - /gitlab-templates/clang-format.yml
       - /gitlab-templates/reuse-lint.yml
-      - /gitlab-templates/linux.yml
       - /gitlab-templates/linux-qt6.yml
-      - /gitlab-templates/freebsd.yml
       - /gitlab-templates/freebsd-qt6.yml
-      - /gitlab-templates/android.yml
       - /gitlab-templates/android-qt6.yml
-      - /gitlab-templates/windows.yml
       - /gitlab-templates/windows-qt6.yml
       - /gitlab-templates/flatpak.yml
diff -urN a/.kde-ci.yml b/.kde-ci.yml
--- a/.kde-ci.yml	2024-03-19 10:39:31.000000000 +0100
+++ b/.kde-ci.yml	2024-03-18 05:20:45.000000000 +0100
@@ -3,26 +3,6 @@
 
 Dependencies:
   - 'on':
-      - Android/Qt5
-      - FreeBSD/Qt5
-      - Linux/Qt5
-      - Windows/Qt5
-    require:
-      frameworks/extra-cmake-modules: '@stable'
-      frameworks/kcoreaddons: '@stable'
-      frameworks/kconfig: '@stable'
-      frameworks/ki18n: '@stable'
-      frameworks/kirigami: '@stable'
-      libraries/kirigami-addons: '@latest'
-  - 'on':
-      - FreeBSD/Qt5
-      - Linux/Qt5
-      - Windows/Qt5
-    require:
-      frameworks/kcrash: '@stable'
-      frameworks/qqc2-desktop-style: '@stable'
-
-  - 'on':
       - Android/Qt6
       - FreeBSD/Qt6
       - Linux/Qt6
diff -urN a/src/qml/ChannelListDelegate.qml b/src/qml/ChannelListDelegate.qml
--- a/src/qml/ChannelListDelegate.qml	2024-03-19 10:39:31.000000000 +0100
+++ b/src/qml/ChannelListDelegate.qml	2024-03-18 05:20:45.000000000 +0100
@@ -4,43 +4,55 @@
 import QtQuick 2.14
 import QtQuick.Controls 2.14 as Controls
 import QtQuick.Layouts 1.14
+import QtQuick.Templates 2.15 as Templates
 import org.kde.TellySkout 1.0
 import org.kde.kirigami 2.19 as Kirigami
 
-Kirigami.SwipeListItem {
-    id: listItem
+Templates.ItemDelegate {
+    id: delegate
 
-    property var listView
     property bool sortable: false
+    property var listView
 
-    actions: [
-        Kirigami.Action {
-            icon.name: checked ? "favorite" : "list-add"
-            text: i18n("Favorite")
-            checkable: true
-            checked: model.channel.favorite
-            onToggled: channelsModel.setFavorite(model.channel.id, checked)
-        }
-    ]
+    width: parent ? parent.width : implicitWidth
+    height: listItem.implicitHeight
 
-    contentItem: RowLayout {
-        Kirigami.ListItemDragHandle {
-            listItem: listItem
-            listView: listItem.listView
-            onMoveRequested: sortable ? listView.model.move(oldIndex, newIndex) : {
+    Kirigami.SwipeListItem {
+        id: listItem
+
+        actions: [
+            Kirigami.Action {
+                icon.name: checked ? "favorite" : "list-add"
+                text: i18n("Favorite")
+                checkable: true
+                checked: model.channel.favorite
+                onToggled: channelsModel.setFavorite(model.channel.id, checked)
             }
-            visible: listItem.sortable
-        }
+        ]
 
-        Kirigami.Icon {
-            source: model.channel.refreshing ? "view-refresh" : model.channel.image === "" ? "tv" : Fetcher.image(model.channel.image)
-        }
+        contentItem: RowLayout {
+            Kirigami.ListItemDragHandle {
+                listItem: listItem
+                listView: delegate.listView
+                onMoveRequested: (oldIndex, newIndex) => {
+                    if (sortable)
+                        listView.model.move(oldIndex, newIndex);
+
+                }
+                visible: delegate.sortable
+            }
+
+            Kirigami.Icon {
+                source: model.channel.refreshing ? "view-refresh" : model.channel.image === "" ? "tv" : Fetcher.image(model.channel.image)
+            }
+
+            Controls.Label {
+                Layout.fillWidth: true
+                height: Math.max(implicitHeight, Kirigami.Units.iconSizes.smallMedium)
+                text: model.channel.displayName || model.channel.name
+                color: listItem.checked || (listItem.pressed && !listItem.checked && !listItem.sectionDelegate) ? listItem.activeTextColor : listItem.textColor
+            }
 
-        Controls.Label {
-            Layout.fillWidth: true
-            height: Math.max(implicitHeight, Kirigami.Units.iconSizes.smallMedium)
-            text: model.channel.displayName || model.channel.name
-            color: listItem.checked || (listItem.pressed && !listItem.checked && !listItem.sectionDelegate) ? listItem.activeTextColor : listItem.textColor
         }
 
     }
diff -urN a/src/qml/ChannelListPage.qml b/src/qml/ChannelListPage.qml
--- a/src/qml/ChannelListPage.qml	2024-03-19 10:39:31.000000000 +0100
+++ b/src/qml/ChannelListPage.qml	2024-03-18 05:20:45.000000000 +0100
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
 import Qt.labs.platform 1.1
-import QtQuick 2.14
+import QtQuick 2.15
 import QtQuick.Controls 2.14 as Controls
 import QtQuick.Layouts 1.14
 import org.kde.TellySkout 1.0
@@ -41,6 +41,7 @@
         anchors.fill: parent
         model: root.onlyFavorites ? channelsModel : proxyModel
         currentIndex: -1 // do not select first list item
+        reuseItems: true
 
         ChannelsProxyModel {
             id: proxyModel
@@ -56,17 +57,7 @@
             onlyFavorites: root.onlyFavorites
         }
 
-        delegate: Kirigami.DelegateRecycler {
-            width: parent ? parent.width : implicitWidth
-            sourceComponent: delegateComponent
-        }
-
-    }
-
-    Component {
-        id: delegateComponent
-
-        ChannelListDelegate {
+        delegate: ChannelListDelegate {
             listView: channelList
             sortable: root.sortable
         }
diff -urN a/src/qml/ChannelTableDelegate.qml b/src/qml/ChannelTableDelegate.qml
--- a/src/qml/ChannelTableDelegate.qml	2024-03-19 10:39:31.000000000 +0100
+++ b/src/qml/ChannelTableDelegate.qml	2024-03-18 05:20:45.000000000 +0100
@@ -49,7 +49,7 @@
         height: (currentTimestamp - program.start) / 60000 * root.pxPerMin
         Component.onCompleted: {
             // update overlay if it is open (for this program)
-            if (root.overlay.sheetOpen && root.overlay.programId === program.id)
+            if (root.overlay.visible && root.overlay.programId === program.id)
                 updateOverlay();
 
         }
diff -urN a/src/qml/ChannelTablePage.qml b/src/qml/ChannelTablePage.qml
--- a/src/qml/ChannelTablePage.qml	2024-03-19 10:39:31.000000000 +0100
+++ b/src/qml/ChannelTablePage.qml	2024-03-18 05:20:45.000000000 +0100
@@ -94,6 +94,7 @@
             contentHeight: 24 * 60 * pxPerMin
             contentWidth: content.width
             boundsBehavior: Flickable.StopAtBounds
+            clip: true
             Component.onCompleted: {
                 // scroll to current time
                 var today = new Date();
diff -urN a/src/qml/GroupListDelegate.qml b/src/qml/GroupListDelegate.qml
--- a/src/qml/GroupListDelegate.qml	2024-03-19 10:39:31.000000000 +0100
+++ b/src/qml/GroupListDelegate.qml	2024-03-18 05:20:45.000000000 +0100
@@ -8,14 +8,13 @@
 import org.kde.kirigami 2.19 as Kirigami
 
 Kirigami.SwipeListItem {
-    leftPadding: 0
-    rightPadding: 0
+    padding: 0
 
-    contentItem: Kirigami.BasicListItem {
+    contentItem: Controls.ItemDelegate {
         anchors.top: parent.top
         anchors.bottom: parent.bottom
         text: model.group.name
-        icon: model.group.refreshing ? "view-refresh" : ""
+        icon.name: model.group.refreshing ? "view-refresh" : ""
         onClicked: {
             lastGroup = model.group.id;
             Fetcher.fetchGroup(model.group.url, model.group.id);
diff -urN a/src/qml/SettingsPage.qml b/src/qml/SettingsPage.qml
--- a/src/qml/SettingsPage.qml	2024-03-19 10:39:31.000000000 +0100
+++ b/src/qml/SettingsPage.qml	2024-03-18 05:20:45.000000000 +0100
@@ -3,7 +3,7 @@
 
 import QtQuick 2.14
 import QtQuick.Controls 2.14 as Controls
-import QtQuick.Dialogs 1.0
+import QtQuick.Dialogs
 import QtQuick.Layouts 1.14
 import QtQuick.Window 2.15
 import org.kde.kirigami 2.19 as Kirigami
@@ -137,11 +137,10 @@
                 id: fileDialog
 
                 nameFilters: [i18n("XML files (*.xml)"), i18n("All files (*)")]
-                selectExisting: true
-                selectMultiple: false
+                fileMode: FileDialog.OpenFile
                 onAccepted: {
                     // remove prefixed "file://"
-                    const path = fileUrl.toString().replace(/^(file:\/{2})/, "");
+                    const path = selectedFile.toString().replace(/^(file:\/{2})/, "");
                     xmltvFile.text = path;
                     _settings.xmltvFile = path;
                 }
diff -urN a/src/qml/TellySkoutGlobalDrawer.qml b/src/qml/TellySkoutGlobalDrawer.qml
--- a/src/qml/TellySkoutGlobalDrawer.qml	2024-03-19 10:39:31.000000000 +0100
+++ b/src/qml/TellySkoutGlobalDrawer.qml	2024-03-18 05:20:45.000000000 +0100
@@ -13,7 +13,7 @@
     actions: [
         Kirigami.Action {
             text: i18n("Favorites")
-            iconName: "view-calendar-day"
+            icon.name: "view-calendar-day"
             onTriggered: {
                 pageStack.layers.clear();
                 pageStack.clear();
@@ -24,7 +24,7 @@
         },
         Kirigami.Action {
             text: i18n("Select Favorites")
-            iconName: "favorite"
+            icon.name: "favorite"
             onTriggered: {
                 pageStack.layers.clear();
                 pageStack.clear();
@@ -33,7 +33,7 @@
         },
         Kirigami.Action {
             text: i18n("Sort Favorites")
-            iconName: "view-sort"
+            icon.name: "view-sort"
             onTriggered: {
                 pageStack.layers.clear();
                 pageStack.clear();
@@ -46,13 +46,13 @@
         },
         Kirigami.Action {
             text: i18n("Settings")
-            iconName: "settings-configure"
+            icon.name: "settings-configure"
             onTriggered: pageStack.layers.push("qrc:/SettingsPage.qml")
             enabled: pageStack.layers.currentItem.title !== i18n("Settings")
         },
         Kirigami.Action {
             text: i18n("About")
-            iconName: "help-about-symbolic"
+            icon.name: "help-about-symbolic"
             onTriggered: pageStack.layers.push(aboutPage)
             enabled: pageStack.layers.currentItem.title !== i18n("About")
         }
